node/pkg/utils/goroutine-pool/pool.go

```diff
@@ -0,0 +1,40 @@
+package goroutine_pool
```

Comment:
lets just make it simply `pool` or `goroutinepool` underscore is not commonly used for package naming in golang. it'd be nice if you can also rename folder name `goroutine-pool` into `pool` or `goroutinepool`

I personally think just `pool` would be okay because there aren't any similar naming or usage within the code, it won't cause a confusion. `pool` is not also a reserved word in golang so it should be okay.